helpers/apps

```diff
@@ -111,3 +111,92 @@ ynh_remove_apps() {
 		done
 	fi
 }
+
+# Spawn a Bash shell with the app environment loaded
+#
+# usage: ynh_spawn_app_shell --app="app"
+#     | arg: -a, --app=     - the app ID
+#
+# examples:
+#   ynh_spawn_app_shell --app="APP" <<< 'echo "$USER"'
+#   ynh_spawn_app_shell --app="APP" < /tmp/some_script.bash
+#
+# Requires YunoHost version 11.0.* or higher, and that the app relies on packaging v2 or higher.
+# The spawned shell will have environment variables loaded and environment files sourced
+# from the app's service configuration file (defaults to $app.service, overridable by the packager with `service` setting).
+# If the app relies on a specific PHP version, then `php` will be aliased that version.
+ynh_spawn_app_shell() {
+        # Declare an array to define the options of this helper.
+        local legacy_args=a
+        local -A args_array=([a]=app=)
+        local app
+        # Manage arguments with getopts
+        ynh_handle_getopts_args "$@"
+
+        # Force Bash to be used to run this helper
+        if [[ ! $0 =~ \/?bash$ ]]
+        then
+            ynh_print_err --message="Please use Bash as shell"
+            exit 1
+        fi
+
+        # Make sure the app is installed
+        local installed_apps_list=($(yunohost app list --output-as json --quiet | jq -r .apps[].id))
+        if [[ " ${installed_apps_list[*]} " != *" ${app} "* ]]
+        then
+            ynh_print_err --message="$app is not in the apps list"
+            exit 1
+        fi
+
+        # Make sure the app is installed
+        if ! id -u "$app" &>/dev/null; then
+            ynh_print_err --message="There is no \"$app\" system user"
+            exit 1
+        fi
+
+        # Make sure the app has an install_dir setting
+        local install_dir=$(ynh_app_setting_get --app=$app --key=install_dir)
+        if [ -z "$install_dir" ]
+        then
+            ynh_print_err --message="$app has no install_dir setting (does it use packaging format >=2?)"
+            exit 1
+        fi
+
+        # Load the app's service name, or default to $app
+        local service=$(ynh_app_setting_get --app=$app --key=service)
+        [ -z "$service" ] && service=$app;
+
+        # Load the Environment variables from the app's service
+        local env_var=`systemctl show $service.service -p "Environment" --value`
+        [ -n "$env_var" ] && export $env_var;
+        export HOME=$install_dir;
+
+        # Force `php` to its intended version
+        local phpversion=$(ynh_app_setting_get --app=$app --key=phpversion)
+        if [ -n "$phpversion" ]
+        then
+            eval "php() { php${phpversion} \"\$@\"; }"
```

Comment:
Nitpicking: I suggest this for more concision and simplicity:
```suggestion
            alias php="php${phpversion}"
```

Reply:
`alias` is not propagated to subshells. ;) The `eval+export` way is the most concise way I found.

Reply:
Interesting! Even `alias+export`?

BTW, if I attempt on a local shell to run `eval "foo() { echo \"$@\"; }"` and then `su -s /bin/bash $USER`, I cannot invoke `foo WHATEVER_ARG`. Wonder what is actually missing to make it work.

Reply:
Even. Tested it earlier this morning, even with `set -a`.

You forgot to escape `$`:

```
bash
eval "foo() { echo \"\$@\"; }"
export -f foo
su -s /bin/bash 
type foo # returns the function definition
foo testme  # returns testme
```